lib/internal/modules/esm/load_modules_in_isolation.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const {
+  ArrayIsArray,
+  ObjectCreate,
+} = primordials;
+
+let debug = require('internal/util/debuglog').debuglog('esm', (fn) => {
+  debug = fn;
+});
+
+
+module.exports = function loadModulesInIsolation(specifiers, loaders = []) {
+  if (!ArrayIsArray(specifiers) || specifiers.length === 0) { return; }
+  debug('loading ESM modules', specifiers, 'with loaders', loaders);
+
+  let cwd;
+  try {
+    cwd = process.cwd() + '/';
+  } catch {
+    cwd = 'file:///';
+  }
+
+  // ! maybe move this outside
+  const { ESMLoader } = require('internal/modules/esm/loader');
```

Comment:
This file can be merged into `esm/loader` and gets exported as part of that `module.exports`, then you no longer need to require that file below. And there would be one less `require()` call from `modules/esm/worker.js` as well.

(And to the comment above - that creates a new `process/esm_loader -> esm/load_modules_in_isolation -> internal/modules/esm/loader -> internal/modules/esm/* -> process/esm_loader` circular dependency if you move it outside, the `process/esm_loader -> internal/modules/esm/loader -> process/esm_loader` is already inevitable, but let's try not to make the mess worse by adding a new circular path, which might work if you are lucky but makes the code harder to reason about especially when there's TDZs around).

Reply:
I need to pivot approach on this PR, and in the pivot, this can go back to where it was.